interfaces/cython/cantera/_onedim.pyx

```diff
@@ -396,24 +399,54 @@ cdef class SymmetryPlane1D(Boundary1D):
 
 cdef class Surface1D(Boundary1D):
     """A solid surface."""
-    def __cinit__(self, *args, **kwargs):
-        self.surf = new CxxSurf1D()
+    def __cinit__(self, _SolutionBase phase, **kwargs):
+        self.surf = new CxxSurf1D(phase._base)
         self.boundary = <CxxBoundary1D*>(self.surf)
 
     def __dealloc__(self):
         del self.surf
 
 
 cdef class ReactingSurface1D(Boundary1D):
-    """A reacting solid surface."""
-    def __cinit__(self, *args, **kwargs):
-        self.surf = new CxxReactingSurf1D()
+    """A reacting solid surface.
+
+    :param phase:
+        The (surface) phase corresponding to the boundary
+
+    .. versionchanged:: 3.0
+
+        Starting in Cantera 3.0, parameter `phase` should reference surface instead of
+        gas phase.
+    """
+    def __cinit__(self, _SolutionBase phase, **kwargs):
+        if isinstance(phase, _SolutionBase) and phase.phase_of_matter != "gas":
+            self.surf = new CxxReactingSurf1D(phase._base)
```

Comment:
I think the `isinstance` check is unnecessary -- the base class `__cinit__` should already catch this, right?